e2e-test-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -38,6 +46,50 @@ public static OpenAiChatCompletionOutput chatCompletion() {
     return OpenAiClient.forModel(GPT_35_TURBO).chatCompletion(request);
   }
 
+  /**
+   * Asynchronous stream of an OpenAI chat request
+   *
+   * @return the emitter that streams the assistant message response
+   */
+  @GetMapping("/streamChatCompletion")
+  @Nonnull
+  public static ResponseEntity<ResponseBodyEmitter> streamChatCompletion() {
+    final var request =
+        new OpenAiChatCompletionParameters()
+            .setMessages(
+                List.of(
+                    new OpenAiChatUserMessage()
+                        .addText(
+                            "Can you give me the first 100 number of the Fibonacci sequence?")));
+
+    ResponseBodyEmitter emitter = new ResponseBodyEmitter();
+    // Cloud SDK's ThreadContext is vital for the request to successfully execute.
+    ThreadContextExecutors.getExecutor()
+        .submit(
+            () -> {
+              // try-with-resources to ensure the OpenAiChatCompletionStream is closed
+              try (var stream = OpenAiClient.forModel(GPT_35_TURBO).streamChatCompletion(request)) {
+                stream
+                    .getDeltaStream()
+                    // The first two and the last delta do not contain any content
+                    .filter(delta -> delta.getDeltaContent() != null)
+                    .forEach(
+                        delta -> {
+                          try {
+                            emitter.send(delta.getDeltaContent());
+                          } catch (IOException e) {
+                            log.error(Arrays.toString(e.getStackTrace()));
+                            emitter.completeWithError(e);
+                          }
+                        });
+              } finally {
+                emitter.complete();
+              }
+            });
+    // MediaType.TEXT_EVENT_STREAM allows the browser to display the content as it is streamed
+    return ResponseEntity.ok().contentType(MediaType.TEXT_EVENT_STREAM).body(emitter);
```

Comment:
this is a problem, because it always returns 200, even when `streamChatCompletion(request)` fails immediately, e.g. with a 400 bad request.

Instead, `ResponseEntity.ok()` should only be invoked, once `streamChatCompletion` got a 200 (and potentially started to get some content). Not sure how to implement this exactly, maybe we can just move the `streamChatCompletion` call out of the thread, and only pass the stream to the thread? But maybe that is awkward with the try-with-resource/finally setup?

Reply:
400 is throwing `OpenAiStreamingHandler:39`, which will be treated by our `ControllerExceptionHandler`

Reply:
> which will be treated by our ControllerExceptionHandler

I am afraid it won't, because it's caught by the try inside the thread and has no impact on the `ok()` response.

Reply:
The try doesn't catch anything, it's a try-with-resources to close the stream.

Reply:
true, the try doesn't catch it, it only closes the stream. But because it's inside the thread the exception will not be thrown in the main thread, so yes moving it should fix it